ci/prow/e2e-hub-spoke-incluster-build

```diff
@@ -24,11 +24,15 @@ hub_api_server=$(kubectl config view -o yaml | yq '.clusters[] | select(.. | .na
 clusteradm join \
   --hub-token ${token} \
   --hub-apiserver ${hub_api_server} \
-  --wait \
   --cluster-name ${MINIKUBE}
+
+# Wait for the join command to succeed
+kubectl wait --for=condition=Established crd managedclusters.cluster.open-cluster-management.io
+timeout 60 sh -c "while ! kubectl get managedcluster ${MINIKUBE} >/dev/null 2>&1; do sleep 2; done"
+timeout 30 sh -c "while ! kubectl get csr -l open-cluster-management.io/cluster-name=${MINIKUBE} 2> /dev/null | grep Pending >/dev/null 2>&1; do sleep 2; done"
```

Comment:
I think using `until` would be more readable. I would also not suppress stdout / stderr output, to make troubleshooting easier:
```suggestion
timeout 60 sh -c "until kubectl get managedcluster ${MINIKUBE}; do sleep 2; done"
timeout 30 sh -c "until kubectl get csr -l open-cluster-management.io/cluster-name=${MINIKUBE} | grep Pending; do sleep 2; done"
```
Also, could we target a specific path for `Pending` instead of grepping?

Reply:
Of course, `until` is quite more readable and I should have conformed to the rest of the script :+1: 

In addition, I replaced `grep`, but since there is no `jsonpath` to indicate that the CR's `condition=Pending` (as it's `Pending` when the `CR.status` is empty), we have no other way to check for this CR's existence other than using `jq` + `select` + `match`. I think that this is still better than just `grep`-ping :)